xarray/backends/h5netcdf_.py

```diff
@@ -16,6 +16,7 @@
 
 class H5NetCDFArrayWrapper(BaseNetCDF4Array):
     def __getitem__(self, key):
+        key = key.to_tuple() if hasattr(key, 'to_tuple') else key
```

Comment:
Can you make a helper function for this line, given that it's repeated a few times?

The other option is to just write `key = key.to_tuple()` or even `key = tuple(key)`, if we are OK assuming that key is already a tuple subclass (which should be OK given that all keys go through `Variable.__getitem__` to get here).

Note that `tuple` already returns the same object if it's argument is a tuple:
```
In [1]: x = (1, 2)

In [2]: y = tuple(x)

In [3]: x is y
Out[3]: True
```